### PR TITLE
#3 chat page basic express server

### PR DESCRIPTION
**Pull request by T1To48**
the structure of the server with the express basic server ,  mainly required npm pkgs and also verry verry basic routes setup.
all finished

### CODE REVIEW

File: server/config/config.env [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+NODE_ENV=development  
+
+PORT=8080          
+
+MONGO_URI=mongodb+srv://tofek98:TDWA5GK4wFYi8kAY@test.imfukcq.mongodb.net/
```
Comment (yishain11) [inline server/config/config.env:6]: don't upload keys or passwords to github!
this should be in gitignore

File: server/server.js [hunk 2]
```diff
@@ -1,3 +1,49 @@
...
+const server = app.listen(
+  PORT,
+  console.log(
+    ` ⭐⭐server is running in ${process.env.NODE_ENV} Mode, & made on port ${PORT} ⭐⭐`
+  )
+);
+
+process.on("unhandledRejection", (err, promise) => {
+  console.log(`😡😡 Error: ${err.message} 😡😡`);
+  server.close(() => process.exit(1));
```
Comment (yishain11) [inline server/server.js:48]: also close db connection

File: server/middleware/asyncHandler.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const asyncHandler = (fn) => (req, res, next) => {
+  Promise.resolve(fn(req, res, next).catch(next));
```
Comment (yishain11) [inline server/middleware/asyncHandler.js:2]: what is the point of this function?

File: package.json [hunk 4]
```diff
@@ -2,10 +2,11 @@
   "name": "arabrew",
   "version": "1.0.0",
   "description": "",
-  "main": "index.js",
+  "main": "./server/server.js",
+  "type": "module",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
-    "prepare": "husky install"
```
Comment (yishain11) [inline package.json:8]: return this

File: client/src/main.jsx [hunk 5]
```diff
@@ -1,10 +1,19 @@
+"use client"
```
Comment (yishain11) [inline client/src/main.jsx:1]: what is this? why is it here?

**Review by yishain11 — CHANGES_REQUESTED** [review]

File: server/config/config.env (hunk 1 above)
Comment (T1To48) [reply server/config/config.env]: FIXED

File: server/middleware/asyncHandler.js (hunk 3 above)
Comment (T1To48) [reply server/middleware/asyncHandler.js:2]: a usage example 
```
```
export const addNew=asyncHandler(async(req,res,next)=>{
    const newTodo=await Todo.create(req.body);
    if (!newTodo){
       return   next(new Error("failed to add todo to the list"))
    }
    else{
        res.status(200).json({
            success:true,

        })
    }
})
```
```
it is a middleware to better configure the controller function's try/catch blocks
resons to use 
less writing , because no need to open try catch block each time we write a controller function , 
next() is responsible of transporting the error to the catch field in the middleware
very good well knows convention in nodejs  i saw davegray and bradtraverssy and others use it 
express has its own asynchandler package , which include ver similar code , 
in conclusion:
easy to read and understand the controllers  functions

File: server/api/user/user.routes.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import express from "express";
+
+const router=express.Router();
+
+// router.route("/")
```
Comment (yishain11) [inline server/api/user/user.routes.js:5]: remove commented out code

File: server/routes.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+import translationRouter from "./api/translation/translation.routes.js";
+
+const router = express.Router();
+
+router.use("/user", userRouter);
+
+router.use("/chat", chatRouter);
+
+router.use("/message", messageRouter);
+
```
Comment (yishain11) [inline server/routes.js:14]: remove empty lines

**Review by yishain11 — CHANGES_REQUESTED** [review]

File: server/server.js [hunk 8]
```diff
@@ -1,3 +1,54 @@
...
+app.use(cors());
+
+if (process.env.NODE_ENV !== "production") {
+  app.use(pino);
+}
+
+app.get("/", (req, res) => {
+  res.send("Server is ACTIVE  ");
+});
+
```
Comment (yishain11) [inline server/server.js:33]: remove '/' endpoint, it does nothing

File: server/api/translation/translation.routes.js [hunk 9]
```diff
@@ -0,0 +1,6 @@
+import express from "express";
+
+const router=express.Router();
+
```
Comment (yishain11) [inline server/api/translation/translation.routes.js:4]: remove empty lines

File: server/api/message/message.routes.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
+import express from "express";
+
+const router=express.Router();
+
```
Comment (yishain11) [inline server/api/message/message.routes.js:4]: remove empty lines

File: server/api/chat/chat.routes.js [hunk 11]
```diff
@@ -0,0 +1,6 @@
+import express from "express";
+
+const router=express.Router();
+
```
Comment (yishain11) [inline server/api/chat/chat.routes.js:4]: remove empty lines

**Review by yishain11 — CHANGES_REQUESTED** [review]
